Better handling of muting on join

sandr3j opened the pull request:
Based on feedback from @ancorgs I improved automatic muting on join.

ancorgs commented on src/app/components/feed/feed-connection.factory.js:
```diff
@@ -64,8 +64,11 @@
-        var cfg = {video: true, audio: false};
+        var cfg = {video: true, audio: true};
         if (this.role === "main") {
+          if (options.muted){
```
Please document the new argument in the heading of the function.

ancorgs commented on src/app/components/notifier/mute-notifier.service.js:
```diff
@@ -27,6 +30,16 @@
     }
 
+    function joinedMuted(){
+      info("You're muted. Unmute yourself to join the conversation!");
```
I would like something more explicit about the reason (and we don't need the "unmute yourself" part because we already have a button saying "unmute"). Something like this (just my first attempt SURE you can improve it):
"Joining to a conversation with more than X participants. You'll start muted."

sandr3j replied:
Yes I thought about "You're muted because there are more than X participants in this room. Unmute yourself..." but I wanted to use a brief sentence to avoid wrapping of the notification on small screens. I will try to make a better sentence.

ancorgs replied:
Yes, wording is usually much harder than coding. :smile:

ancorgs commented:
Wow! Quite impressive but please open separate pull requests for separate features/fixes.

I will only comment here stuff from the start muted feature.

The commit to fix the newVal problem should disappear (we already have a pull request for it)

The commits for push to talk should be moved to a different pull request so we can have a focused conversation there.

Thanks!

ancorgs commented on src/app/components/feed/jh-feed.directive.js:
```diff
@@ -61,6 +61,10 @@
             mutedWarningTimeout = now();
           });
+          scope.$on('muted.Join', function() {
+            mutedWarningTimeout = secondsFromNow(60);
```
Wouldn't it be better to set it to `now()` or to `secondsFromNow(3)` ? It would have a similar effect but with more consistent times (the same time than muting myself or the same one that having been muted). Isn't it?

sandr3j replied:
The reasoning behind this was to stop a second notification (when voice is detected) from appearing while the first is not dismissed yet. But now I see that it didn't fix it properly so I will take another approach..

ancorgs commented:
Parts about starting muted already commented. Only some nitpicking details here and there, but in general it looks really nice. Great job!

Waiting for a clean pull request to merge that part and release a new version.

sandr3j commented:
Thanks for the kind words, great that you like it :)
I added now new commits and cleaned up this pull request. I will make new pull request for push-to-talk

ancorgs commented:
Thanks for the changes. It looks nice now. Let's merge.
